### PR TITLE
TrainingPage-API-Integration

### PR DESCRIPTION
**Pull request by athulmuraleedharan**

### CODE REVIEW

File: client/idp/src/Types/Dto/training.dto.d.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+declare interface TrainingViewDto{
+    Id: number,
+     Name : string,
```
Comment (mishelshaji) [inline client/idp/src/Types/Dto/training.dto.d.ts:3]: Format the code.

File: client/idp/src/Types/Dto/training.dto.d.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+declare interface TrainingViewDto{
```
Comment (mishelshaji) [inline client/idp/src/Types/Dto/training.dto.d.ts:1]: Check naming convention.

**Review by mishelshaji — CHANGES_REQUESTED** [review]

File: client/idp/src/Types/Dto/training.dto.d.ts (hunk 1 above)
Comment (athulmuraleedharan) [reply client/idp/src/Types/Dto/training.dto.d.ts:3]: corrected

File: client/idp/src/Types/Dto/training.dto.d.ts (hunk 2 above)
Comment (athulmuraleedharan) [reply client/idp/src/Types/Dto/training.dto.d.ts:1]: corrected

File: client/idp/src/app/shared/objective-detailed/objective-detailed.component.html [hunk 3]
```diff
@@ -19,36 +19,21 @@ <h5 class="text-secondary">Training Item</h5>
...
-                        <td><input type="text" id="training-progress"></td>
-                        <td>
-                            <button class="btn btn-warning" (click)="submitProgress()">
-                                <span class="fa fa-eye"></span>
-                                Update
-                            </button>
-                        </td>
+                <tbody>
+                    <tr *ngFor="let training of trainings">
+                        <td>{{training.Id}}</td>
```
Comment (mishelshaji) [inline client/idp/src/app/shared/objective-detailed/objective-detailed.component.html:24]: use camelCasing for property values.

File: client/idp/src/app/shared/objective-detailed/objective-detailed.component.html [hunk 4]
```diff
@@ -19,36 +19,21 @@ <h5 class="text-secondary">Training Item</h5>
...
+                        <td>{{training.Name}}</td>
```
Comment (mishelshaji) [inline client/idp/src/app/shared/objective-detailed/objective-detailed.component.html:25]: use camelCasing for property values.

File: client/idp/src/app/shared/objective-detailed/objective-detailed.component.html [hunk 5]
```diff
@@ -19,36 +19,21 @@ <h5 class="text-secondary">Training Item</h5>
...
+                        <td>{{training.Link}}</td>
```
Comment (mishelshaji) [inline client/idp/src/app/shared/objective-detailed/objective-detailed.component.html:26]: use camelCasing for property values.

File: client/idp/src/app/shared/objective-detailed/objective-detailed.component.html [hunk 6]
```diff
@@ -19,36 +19,21 @@ <h5 class="text-secondary">Training Item</h5>
...
+                        <td>{{training.Progress}}</td>
```
Comment (mishelshaji) [inline client/idp/src/app/shared/objective-detailed/objective-detailed.component.html:27]: use camelCasing for property values.

File: client/idp/src/app/shared/objective-detailed/objective-detailed.component.html [hunk 7]
```diff
@@ -19,36 +19,21 @@ <h5 class="text-secondary">Training Item</h5>
...
+                        <td>{{training.StartDate}}</td>
```
Comment (mishelshaji) [inline client/idp/src/app/shared/objective-detailed/objective-detailed.component.html:28]: use camelCasing for property values.

File: client/idp/src/app/shared/objective-detailed/objective-detailed.component.html [hunk 8]
```diff
@@ -19,36 +19,21 @@ <h5 class="text-secondary">Training Item</h5>
...
+                        <td>{{training.EndDate}}</td>
```
Comment (mishelshaji) [inline client/idp/src/app/shared/objective-detailed/objective-detailed.component.html:29]: use camelCasing for property values.

File: client/idp/src/app/shared/objective-detailed/objective-detailed.component.html [hunk 9]
```diff
@@ -19,36 +19,21 @@ <h5 class="text-secondary">Training Item</h5>
...
+                        <td>{{training.UserId}}</td>
```
Comment (mishelshaji) [inline client/idp/src/app/shared/objective-detailed/objective-detailed.component.html:30]: use camelCasing for property values.

File: client/idp/src/app/shared/objective-detailed/objective-detailed.component.html [hunk 10]
```diff
@@ -19,36 +19,21 @@ <h5 class="text-secondary">Training Item</h5>
...
+                        <td>{{training.IdpId}}</td>
```
Comment (mishelshaji) [inline client/idp/src/app/shared/objective-detailed/objective-detailed.component.html:31]: use camelCasing for property values.

File: client/idp/src/app/shared/objective-detailed/objective-detailed.component.html [hunk 11]
```diff
@@ -19,36 +19,21 @@ <h5 class="text-secondary">Training Item</h5>
...
+                        <td>{{training.ObjectiveId}}</td>
```
Comment (mishelshaji) [inline client/idp/src/app/shared/objective-detailed/objective-detailed.component.html:32]: use camelCasing for property values.

File: client/idp/src/app/shared/objective-detailed/objective-detailed.component.ts [hunk 12]
```diff
@@ -6,10 +6,10 @@ import { CommonModule } from '@angular/common';
   standalone: true,
   imports: [CommonModule],
   templateUrl: './objective-detailed.component.html',
-  styleUrls: ['./objective-detailed.component.css']
+  styleUrls: ['./objective-detailed.component.css'],
 })
 export class ObjectiveDetailedComponent {
-  list: number[] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 1, 2, 3];
+  trainings: TrainingViewDto[] = [];
   edit: boolean = false;
```
Comment (mishelshaji) [inline client/idp/src/app/shared/objective-detailed/objective-detailed.component.ts:13]: Remove the type specification. Also, add a blank line above.

File: client/idp/src/app/user/user-routing.module.ts [hunk 13]
```diff
@@ -24,17 +26,17 @@ const routes: Routes = [
...
       { path: 'add-training', component: AddTrainingComponent },
-      { path: 'add-action', component: AddActionComponent},
+      { path: 'add-action', component: AddActionComponent },
       { path: 'home', component: HomepageComponent },
       { path: 'objective-detailed', component: ObjectiveDetailedComponent },
       { path: 'idp', component: ShowAllIdpComponent },
-      { path: 'objective', component: CreateObjectiveComponent }
-    ]
-  }
+      { path: 'objective', component: CreateObjectiveComponent },
```
Comment (mishelshaji) [inline client/idp/src/app/user/user-routing.module.ts:33]: Remove the trailing comma.

File: client/idp/src/app/user/user-routing.module.ts [hunk 14]
```diff
@@ -24,17 +26,17 @@ const routes: Routes = [
...
+    ],
+  },
 ];
 
 @NgModule({
   imports: [RouterModule.forChild(routes)],
-  exports: [RouterModule]
+  exports: [RouterModule],
```
Comment (mishelshaji) [inline client/idp/src/app/user/user-routing.module.ts:40]: Remove the trailing comma.

**Review by mishelshaji — APPROVED** [review]

**Pull request merged**